test/bdd/steps/datalayer.js

```diff
@@ -0,0 +1,99 @@
+
+    const { dc } = this.state;
+    let data;
+    const myApiImportInfo = await httpApiHelper.apiImportInfo(dc.state.node_rpc_url, this.state.lastImport.data_set_id);
+
+    expect(
+        utilities.findVertexIdValue(myApiImportInfo.import.vertices, 'IDENTIFIER', 'urn:ot:object:actor:id:Company_Green', 'uid', 'urn:ot:object:actor:id:Company_Green:2018-01-01T01:00:00.000-04:00Z-04:00').length,
+        'There should be at least one such vertex',
+    ).to.be.above(0);
+    data = {
+        parent_id: 'urn:epc:id:sgln:Building_Green',
+    };
+    expect(
+        utilities.findVertexUid(myApiImportInfo.import.vertices, 'LOCATION', 'urn:ot:object:actor:id:Company_Green', 'urn:epc:id:sgln:Building_Green_V2', data).length,
+        'There should be at least one such vertex',
+    ).to.be.above(0);
+    data = {
+        category: 'Company',
+        name: 'Green',
+        object_class_id: 'Actor',
+        wallet: '0xBbAaAd7BD40602B78C0649032D2532dEFa23A4C0',
+    };
+    expect(
+        utilities.findVertexUid(myApiImportInfo.import.vertices, 'ACTOR', 'urn:ot:object:actor:id:Company_Green', 'urn:ot:object:actor:id:Company_Green', data).length,
+        'There should be at least one such vertex',
+    ).to.be.above(0);
+    data = {
+        category: 'Beverage',
+        description: 'Wine Bottle',
+        object_class_id: 'Product',
+    };
+    expect(
+        utilities.findVertexUid(myApiImportInfo.import.vertices, 'PRODUCT', 'urn:ot:object:actor:id:Company_Green', 'urn:ot:object:product:id:Product_1', data).length,
+        'There should be at least one such vertex',
+    ).to.be.above(0);
+    data = {
+        expirationDate: '2020-31-12T00:01:54Z',
+        parent_id: 'urn:ot:object:product:id:Product_1',
+        productId: 'urn:ot:object:product:id:Product_1',
+        productionDate: '2017-31-12T00:01:54Z',
+        quantities: {
+            'urn:ot:object:actor:id:Company_Green:2018-01-01T01:00:00.000-04:00Z-04:00': {
+                PCS: '5d3381241af6b16260f680059e9042',
+            },
+        },
+    };
+    expect(
+        utilities.findVertexUid(myApiImportInfo.import.vertices, 'BATCH', 'urn:ot:object:actor:id:Company_Green', 'urn:epc:id:sgtin:Batch_1', data).length,
+        'There should be at least one such vertex',
+    ).to.be.above(0);
+    expect(
+        utilities.findVertexIdValue(myApiImportInfo.import.vertices, 'IDENTIFIER', 'urn:ot:object:actor:id:Company_Green', 'uid', 'urn:epc:id:sgln:Building_Green').length,
+        'There should be at least one such vertex',
+    ).to.be.above(0);
+});
+
+Then(/^DC manually calculated datasets data and root hashes matches ones from blockchain$/, async function () {
+    expect(!!this.state.dc, 'DC node not defined. Use other step to define it.').to.be.equal(true);
+
+    const { dc } = this.state;
+    const myApiImportsInfo = await httpApiHelper.apiImportsInfo(dc.state.node_rpc_url);
+
+    for (const i in Array.from({ length: myApiImportsInfo.length })) {
```

Comment:
Why like this?

Reply:
This for loop actually waits for awaits inside it. 
Regular forEach, for some reason does not wait for awaits